app/src/main/java/com/keylesspalace/tusky/viewmodel/AccountViewModel.kt

```diff
@@ -243,37 +219,35 @@ class AccountViewModel @Inject constructor(
                         else -> {
                         }
                     }
-
-                } else {
+                },
+                {
                     relationshipData.postValue(Error(relation))
                 }
+        )
+                .autoDispose()
+    }
 
-            }
-
-            override fun onFailure(call: Call<Relationship>, t: Throwable) {
-                relationshipData.postValue(Error(relation))
-            }
-        }
-
-        val call = when (relationshipAction) {
-            RelationShipAction.FOLLOW -> mastodonApi.followAccount(accountId, parameter ?: true)
-            RelationShipAction.UNFOLLOW -> mastodonApi.unfollowAccount(accountId)
-            RelationShipAction.BLOCK -> mastodonApi.blockAccount(accountId)
-            RelationShipAction.UNBLOCK -> mastodonApi.unblockAccount(accountId)
-            RelationShipAction.MUTE -> mastodonApi.muteAccount(accountId, parameter ?: true)
-            RelationShipAction.UNMUTE -> mastodonApi.unmuteAccount(accountId)
-        }
-
-        call.enqueue(callback)
-        callList.add(call)
-
+    fun noteChanged(newNote: String) {
+        noteSaved.postValue(false)
+        noteDisposable?.dispose()
+        noteDisposable = Single.timer(1500, TimeUnit.MILLISECONDS)
```

Comment:
This very much looks like some `throttle`, I think RxAndroid has a way to get a stream from a field?

Reply:
You mean something like [TextViewTextChangesObservable](https://github.com/JakeWharton/RxBinding/blob/master/rxbinding/src/main/java/com/jakewharton/rxbinding4/widget/TextViewTextChangesObservable.kt)? It is part of RxBinding which we currently don't have in the project, but I can add it if you think its better.

Reply:
Oh, I thought we either had the library or had a home-grown alternative but maybe I'm mistaking it with something

Reply:
Maybe we should at least modify our `addTextChangedListener` from `ViewExtensions` to allow unsubscribing and use that instead?